src/vivarium/framework/results/manager.py

```diff
@@ -69,6 +71,33 @@ def setup(self, builder: "Builder"):
 
         builder.value.register_value_modifier("metrics", self.get_results)
 
+    def on_post_setup(self, event: Event):
+        # update self._metrics to have all output keys
+        def create_measure_specific_keys(measure: str, stratifications: List[str]) -> None:
+            measure_str = f"MEASURE_{measure}_"
+            individual_stratification_strings = [
+                [
+                    f"{stratification.name.upper()}_{category}"
+                    for category in stratification.categories
+                ]
+                for stratification in sorted(
+                    self._results_context.stratifications, key=lambda x: x.name
+                )
+                if stratification.name in stratifications
+            ]
+            for complete_stratifications in itertools.product(
+                *individual_stratification_strings
+            ):
+                key = measure_str + "_".join(complete_stratifications)
+                self._metrics[key] = 0
+
+        for event in self._results_context.observations:
+            for (_, stratifications), observations in self._results_context.observations[
+                event
+            ].items():
+                for measure, *_ in observations:
+                    create_measure_specific_keys(measure, stratifications)
+
```

Comment:
Why did you choose this approach, initializing everything effectively to zero, rather than simply recording 0 when the filtered pop is 0 (i.e., [change this code](https://github.com/ihmeuw/vivarium/blob/9bde61308d205a9f71a16c3d7ac9cd8b9454201b/src/vivarium/framework/results/context.py#L120-L121))? This seems like potentially a waste of computation at the start of the sim.

Reply:
There are two reasons for this:
- We only have to do this once during setup, rather than during every observation
- It is also possible that the else branch will return some but not all keys, which means we'd need to check there as well